SCClassLibrary/Common/UnitTesting/UnitTest.sc

```diff
@@ -46,31 +47,97 @@ UnitTest {
 		}
 	}
 
-	// run a single test in the name format "TestPolyPlayerPool:test_prepareChildrenToBundle"
-	*runTest { | methodName |
-		var class, method, unitTest;
-		# class, method = methodName.split($:);
-		class = class.asSymbol.asClass;
-		method.asSymbol;
-		method = class.findMethod(method.asSymbol);
-		if(method.isNil) { Error("Test method not found "+methodName).throw };
-		class.new.runTestMethod(method);
+	// run a class or single test in the name format
+	// "TestPolyPlayerPool:test_prepareChildrenToBundle"
+	*runTest { | testItem |
+		var method, failString;
+		case { testItem.isKindOf(Class) } {
+			if (testItem.isKindOf(Meta_UnitTest)) {
+				testItem.run(false, false)
+			} {
+				failString = "%: testItem % is not a subclass of UnitTest."
+			}
+		} { testItem.isKindOf(String) } {
+			method = this.findTestMethodFor(testItem);
+			if(method.notNil) {
+				method.ownerClass.new.runTestMethod(method)
+			} {
+				failString = "%: testItem % is not a valid testMethod string such as %."
+			}
+		} {
+			failString = "%: testItem % is neither a subclass of UnitTest,"
+			" nor a valid test-method string such as %."
+		};
+
+		if (failString.notNil) {
+			failString.format(thisMethod, testItem.cs, "TestABC:test_xyz".cs).warn
```

Comment:
this should be an error.

Reply:
Fine, will do. 
Do you mean throw an error and stop the entire test run there, 
or just post the error as it happens and continue to run the remaining tests?

Reply:
The first